Update dependencies

lionel-faber opened the pull request:
Update dependencies to resolve security vulnerabilities.

lionel-faber committed:
fix/web_api_playground: Update dependencies

- Update dependencies to resolve vulnerabilities
- Update the gulpfile to support gulp 4

maidsafe-highfive commented:
Thanks for the pull request, and welcome! The MaidSafe team is excited to review your changes, and you should hear from @krishnaIndia (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [CONTRIBUTOR.md](../blob/master/CONTRIBUTOR) for more information.

hunterlester commented on safe_web_api_playground/package.json:
```diff
@@ -21,20 +21,19 @@
     "envfile": "^2.1.1",
     "express": "^4.15.3",
-    "gulp": "^3.9.1",
+    "gulp": "^4.0.0",
```
The major version change breaks `gulp.start` operations in `gupfile.js`. Needs updating.

hunterlester commented on safe_web_api_playground/package.json:
```diff
@@ -21,20 +21,19 @@
     "gulp-watch": "^4.3.11",
     "jquery": "^3.2.1",
     "npm-font-open-sans": "^1.1.0",
-    "safe_ux_guidelines": "https://github.com/maidsafe/safe_ux_guidelines.git",
```
In addition to removing `safe_ux_guidelines`, those areas in the app which relied on it need to be refactored such as `index.html` and the local `scss` files.

lionel1704 replied:
From what I observed in the gulpfile the `safe-styles` gulp task inserts the safe styles from the UX guidelines repo into the `static/scss` folder on which the `sass()` operation is performed to transform them into css files which are linked from `index.html`.

hunterlester reviewed (changes requested):
Styles are broken and gulp is not functioning as expected.
If you want I can pull your branch and open up a pull request with the additional changes needed.

lionel1704 commented:
> Styles are broken and gulp is not functioning as expected.
That's strange. They seemed okay to me. Here are the steps I followed.
- Removed `node_modules` inside the web API playground
- Ran `yarn` to install the packages
- `yarn gulp` served the app on `localhost:3003`
I even uploaded the build dir via WHM and the published sited seemed fine too.
Am I going wrong somewhere?

hunterlester commented:
@lionel1704 Most likely, is that yarn has cached `safe_ux_guidelines`, so it may be continuing to install for you even though the repo has been deleted.

lionel1704 commented:
I did a `yarn cache clean` and followed the same steps again. Still seems okay to me. @manavbp gave it a try on his machine and it seemed okay there too. 

If it's still breaking for you, feel free to raise a PR to my branch.

hunterlester commented:
@lionel1704 Okay I'll open a PR on your branch.
This is what my styles look like:
![image](https://user-images.githubusercontent.com/10050403/54702777-c50dc380-4af4-11e9-84f8-f626dc7e4cfe.png)

What are the contents of you `build` directory?

lionel1704 commented:
Here you go [build.zip](https://github.com/maidsafe/safe_examples/files/2989272/build.zip)

hunterlester commented:
@lionel1704 Thank you. `main.css` still contains `scss` from the no longer existing `safe_ux_guidelines`.
Did you delete build directory and then run `yarn gulp` again?

lionel-faber committed:
fix/web_api_playground: fix broken styles

- Remove scss paths from .gitignore
- Add scss files from the safe-ux-guidelines repository
- Remove npm-font-open-sans dependency

hunterlester committed:
chore/build: updates gulpfile to comply with gulp v4.0

hunterlester commented:
@lionel1704 Looking good! Thanks, Lionel.
Last thing is to update the gulpfile: https://github.com/lionel1704/safe_examples/pull/1

lionel-faber committed:
chore/build: updates gulpfile to comply with gulp v4.0

lionel-faber committed:
chore/build: add test script to package.json

lionel1704 commented:
@S-Coyle can we have this merged please?
